scripts/OPNFV/NetworkServices/Methods.class/__methods__/create_networks_and_subnets.rb

```diff
@@ -1,40 +1,89 @@
-exit MIQ_OK # remove this if you wish to create networks
-
-require 'fog/openstack'
-begin
-  tenant_name = 'admin'
-  $evm.log(:info, "Tenant name: #{tenant_name}")
+def get_networks_template(network_service, parent_service)
+  # Get all networks for this service and create a Heat template to represent them
+  
+  template = nil
+  template_content = {}
+  
+  template_content['heat_template_version'] = '2013-05-23'
+  template_content['resources'] = {}
+  
+  network_service.direct_service_children.detect { |x| x.name == 'VNF Networks' }.direct_service_children.each do |vnf_network|
+    
+    vnf_network_properties = JSON.parse(vnf_network.custom_get('properties'))
+    cidr = vnf_network_properties['cidr']
+    
+    network_name = "#{parent_service.name}_#{vnf_network.name}_net"
+    
+    template_content['resources'][network_name] = {}
+    template_content['resources'][network_name]['properties'] = {}
+    template_content['resources'][network_name]['properties']['name'] = "#{vnf_network.name}"
+    template_content['resources'][network_name]['type'] = 'OS::Neutron::Net'
+    
+    subnet_name = "#{parent_service.name}_#{vnf_network.name}_subnet"
+    
+    template_content['resources'][subnet_name] = {}
+    template_content['resources'][subnet_name]['properties'] = {}
+    template_content['resources'][subnet_name]['properties']['name'] = "#{vnf_network.name}_subnet"
+    template_content['resources'][subnet_name]['properties']['cidr'] = cidr
+    template_content['resources'][subnet_name]['properties']['network_id'] = {}
+    template_content['resources'][subnet_name]['properties']['network_id']['get_resource'] = network_name
+    template_content['resources'][subnet_name]['type'] = 'OS::Neutron::Subnet'
+  end
+  
+  vnf_networks_template_name = "#{parent_service.name} #{Time.now.to_i} networks"
+  
+  template = $evm.vmdb('orchestration_template_hot').create(
+    :name      => vnf_networks_template_name, 
+      :orderable => true, 
+      :content   => YAML.dump(template_content))
+  template
+end
 
-  ems = $evm.vmdb('ManageIQ_Providers_Openstack_CloudManager', $evm.root.attributes['dialog_openstack_manager'])
+def deploy_networks(network_service, parent_service)
   
-  raise "ems not found" if ems.nil?
+  network_orchestration_manager = $evm.vmdb('ManageIQ_Providers_Openstack_CloudManager').find_by_name("openstack-nfvpe")
+  networks_template = get_networks_template(network_service, parent_service)
+  networks_orchestration = deploy_networks_stack(network_orchestration_manager, parent_service, networks_template)
+  
+  while networks_orchestration.orchestration_stack_status[0] == 'transient'
```

Comment:
I'll show you later how to do a proper automate loop, with limited retries, should be ok for now :-)